### ISSUE TITLE
FN S3927: No issue is raised when attribute is placed on a lambda

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#5984]
### Description

No issue is raised when attribute is placed on a lambda

### Repro steps

```csharp
public void Bar(IEnumerable<int> collection)
{
    _ = collection.Select([OnSerialized] (x) => x + 1);  // FN

    Action a = [OnDeserializing] () => { }; //  FN

    Action x = true
                   ? ([OnDeserialized] () => { }) //  FN
                   : [OnDeserialized] () => { };  //  FN
}
```

### Related information

* C#/VB.NET Plugins version: 8.43.0.51858
